taca/illumina/__init__.py

```diff
@@ -0,0 +1,172 @@
+        cl = [cl_options.get('XTen')]
+
+        # Main options
+        if cl_options.get('runfolder'):
+            cl.extend(['--runfolder', cl_options.get('runfolder')])
+        cl.extend(['--output-dir', cl_options.get('output-dir', 'Demultiplexing')])
+
+        # Advanced options
+        if cl_options.get('input-dir'):
+            cl.extend(['--input-dir', cl_options.get('input-dir')])
+        if cl_options.get('intensities-dir'):
+            cl.extend(['--intensities-dir', cl_options.get('intensities-dir')])
+        if cl_options.get('interop-dir'):
+            cl.extend(['--interop-dir', cl_options.get('interop-dir')])
+        if cl_options.get('stats-dir'):
+            cl.extend(['--stats-dir', cl_options.get('stats-dir')])
+        if cl_options.get('reports-dir'):
+            cl.extend(['--reports-dir', cl_options.get('reports-dir')])
+
+        # Processing cl_options
+        threads = cl_options.get('loading-threads')
+        if threads and type(threads) is int:
+            cl.extend(['--loading-threads', '{}'.format(threads)])
+        threads = cl_options.get('demultiplexing-threads')
+        if threads and type(threads) is int:
+            cl.extend(['--demultiplexing-threads', '{}'.format(threads)])
+        threads = cl_options.get('processing-threads')
+        if threads and type(threads) is int:
+            cl.extend(['--processing-threads', '{}'.format(threads)])
+        threads = cl_options.get('writing-threads')
+        if threads and type(threads) is int:
+            cl.extend(['--writing-threads', '{}'.format(threads)])
+
+        # Behavioral options
+        adapter_stringency = cl_options.get('adapter-stringency')
+        if adapter_stringency and type(adapter_stringency) is float:
+            cl.extend(['--adapter-stringency', adapter_stringency])
+        aggregated_tiles = cl_options.get('aggregated-tiles')
+        if aggregated_tiles and aggregated_tiles in ['AUTO', 'YES', 'NO']:
+            cl.etend(['--aggregated-tiles', aggregated_tiles])
+        barcode_missmatches = cl_options.get('barcode-missmatches')
+        if barcode_missmatches and type(barcode_missmatches) is int \
+                and barcode_missmatches in range(3):
+            cl.extend(['--barcode-missmatches', barcode_missmatches])
+        if cl_options.get('create-fastq-for-index-reads'):
+            cl.append('--create-fastq-for-index-reads')
+        if cl_options.get('ignore-missing-bcls'):
+            cl.append('--ignore-missing-bcls')
+        if cl_options.get('ignore-missing-filter'):
+            cl.append('--ignore-missing-filter')
+        if cl_options.get('ignore-missing-locs'):
+            cl.append('--ignore-missing-locs')
+        mask = cl_options.get('mask-short-adapter-reads')
+        if mask and type(mask) is int:
+            cl.extend(['--mask-short-adapter-reads', mask])
+        minimum = cl_options.get('minimum-trimmed-reads')
+        if minimum and type(minimum) is int:
```

Comment:
Same here :) no point of running with `0`

but good catch @mariogiov (in coding aspect) :clap: